plugins/modules/gpg_keypair.py

```diff
@@ -0,0 +1,507 @@
+                subkey_count += 1
+                if subkey_count > len(params['subkeys']):
+                    break
+
+                subkey = re.search(r'.+:([0-9]+):([0-9]+):[0-9A-Z]+:[0-9]+:+([a-z]+):+\+:+([0-9a-zA-Z]+):', line)
+                subkey_type = key_type_from_algo(int(subkey.group(2)))
+                subkey_length = int(subkey.group(1))
+                subkey_curve = subkey.group(4)
+                subkey_usage = expand_usages(subkey.group(3))
+
+                if params['subkeys'][subkey_count]['type'] and params['subkeys'][subkey_count]['type'] != subkey_type:
+                    break
+                elif params['subkeys'][subkey_count]['usage'] and tuple(params['subkeys'][subkey_count]['usage']) not in itertools.permutations(subkey_usage):
+                    break
+                elif subkey_type in ['RSA', 'DSA', 'ELG']:
+                    if params['subkeys'][subkey_count]['length'] and params['subkeys'][subkey_count]['length'] != subkey_length:
+                        break
+                else:
+                    if params['subkeys'][subkey_count]['curve'] and params['subkeys'][subkey_count]['curve'] != subkey_curve:
+                        break
+        else:
+            if uid_present:
+                matching_keys.append(fingerprint)
+
+    if params['state'] == 'present':
+        result = generate_keypair(module, params, matching_keys, check_mode)
+    else:
+        result = delete_keypair(module, matching_keys, check_mode)
+    return result
+
+
+def main():
+    key_types = ['RSA', 'DSA', 'ECDSA', 'EDDSA', 'ECDH', 'ELG']
+    key_curves = ['nistp256', 'nistp384', 'nistp521', 'brainpoolP256r1', 'brainpoolP384r1', 'brainpoolP512r1', 'secp256k1', 'ed25519', 'cv25519']
+    key_usages = ['encr', 'sign', 'auth', 'cert']
+
+    module = AnsibleModule(
+        argument_spec=dict(
+            state=dict(type='str', default='present', choices=['present', 'absent']),
+            key_type=dict(type='str', choices=key_types[:-2]),
+            key_length=dict(type='int', no_log=False),
+            key_curve=dict(type='str', choices=key_curves[:-1]),
+            key_usage=dict(type='list', elements='str', default=[], choices=key_usages),
+            subkeys=dict(
+                type='list',
+                elements='dict',
+                no_log=False,
+                default=[],
+                options=dict(
+                    subkey_type=dict(type='str', choices=key_types),
+                    subkey_length=dict(type='int', no_log=False),
+                    subkey_curve=dict(type='str', choices=key_curves),
+                    subkey_usage=dict(type='list', elements='str', default=[], choices=key_usages[:-1])
+                ),
+                required_if=[
+                    ['subkey_type', 'ECDSA', ['subkey_curve']],
+                    ['subkey_type', 'EDDSA', ['subkey_curve']],
+                    ['subkey_type', 'ECDH', ['subkey_curve']]
+                ]
```

Comment:
Generally it's a good idea to always use trailing commas. This will reduce the size of diffs when this is ever changed.
```suggestion
                    ['subkey_type', 'ECDH', ['subkey_curve']],
                ],
```